voluptuous/validators.py

```diff
@@ -192,12 +192,13 @@ class _WithSubValidators(object):
     def __init__(self, *validators, **kwargs):
         self.validators = validators
         self.msg = kwargs.pop('msg', None)
+        self.required = kwargs.pop('required', False)
 
     def __voluptuous_compile__(self, schema):
-        self._compiled = [
-            schema._compile(v)
-            for v in self.validators
-        ]
+        self._compiled = []
+        for v in self.validators:
+            schema.required = self.required
```

Comment:
Is this still a leftover? Should this be `v.required = self.required`?

Reply:
No, it is correct. Test fail otherwise

Also note that the line must be inside the loop. I'm not totally clear on what happen... but tests are passing ...